subprojects/dependency-management/src/main/java/org/gradle/api/internal/notations/DependencyMapNotationConverter.java

```diff
@@ -44,8 +44,15 @@ protected T parseMap(@MapKey("group") @Optional String group,
                          @MapKey("configuration") @Optional String configuration,
                          @MapKey("ext") @Optional String ext,
                          @MapKey("classifier") @Optional String classifier) {
-        T dependency = instantiator.newInstance(resultingType, group, name, version, configuration);
-        ModuleFactoryHelper.addExplicitArtifactsIfDefined(dependency, ext, classifier);
+        T dependency;
```

Comment:
⭕️ Why did this change?

Reply:
I made the type parameter of `DependencyMapNotationConverter` unbound so that we can reuse it to parse map notations into `ComponentDependencyMetadataDetailsAdapter`. However that class does not have a constructor that takes a configuration as we do not support configurations in component metadata rules. Since we might never support them there, I  did not want to introduce this constructor.